ytmusicapi/parsers/browsing.py

```diff
@@ -29,15 +29,56 @@ def parse_search_result(self, data, resultType=None):
             search_result['title'] = get_item_text(data, 0)
 
         if resultType in ['artist', 'album', 'playlist']:
-            search_result['browseId'] = nav(data, NAVIGATION_BROWSE_ID)
+            browse_id = nav(data, NAVIGATION_BROWSE_ID, True)
+            if not browse_id:  # song result
+                search_result['videoId'] = nav(
+                    data, PLAY_BUTTON)['playNavigationEndpoint']['watchEndpoint']['videoId']
+                search_result['playlistId'] = nav(
+                    data, PLAY_BUTTON)['playNavigationEndpoint']['watchEndpoint']['playlistId']
+                flex_items = [
+                    nav(get_flex_column_item(data, i), ['text', 'runs'], True) for i in range(4)
+                ]
+                if flex_items[0]:
+                    search_result['title'] = {flex_items[0][0]['text']}
+                if flex_items[1]:
+                    search_result['artists'] = [{
+                        'name': flex_items[1][0]['text'],
+                        'id': nav(flex_items[1][0], NAVIGATION_BROWSE_ID)
+                    }]
+                if flex_items[2]:
+                    search_result['album'] = {
+                        'name': flex_items[2][0]['text'],
+                        'id': nav(flex_items[2][0], NAVIGATION_BROWSE_ID)
+                    }
+                if flex_items[3]:
+                    search_result['duration'] = flex_items[3][0]['text']
+                search_result['resultType'] = 'song'
+            else:  # artist or album result
+                search_result['browseId'] = browse_id
+                if 'artist' in search_result['browseId']:
+                    search_result['resultType'] = 'artist'
+                else:
+                    flex_items2 = [
+                        nav(get_flex_column_item(data, i), ['text', 'runs'], True)
+                        for i in range(4)
+                    ]
+                    if flex_items2[0]:
+                        search_result['title'] = flex_items2[0][0]['text']
+                    if flex_items2[2]:
+                        search_result['artist'] = {'name': flex_items2[2][0]['text'], 'id': None}
+                    if flex_items2[3]:
+                        search_result['releaseDate'] = flex_items2[3][0]['text']
+                    search_result['resultType'] = 'album'
```

Comment:
What is the reason for adding this code? Seems it's only copied from further down?

IMO there is no need to change the parser at all, as the continuation items should have an identical format. Feel free to correct me if I'm missing something here

Reply:
Apologies I never got back to you on this - real life got in the way a bit! Did you end up needing to change the parser as I had thought? 

Thanks for implementing, in any event - I might switch mopidy-youtube over to your implementation rather than the one I'm currently - yours seems very robust. Or possibly make a proper mopidy-ytmusic, with which includes vthe authenticated functions, too.

Anyway, thanks again!